frontend/internal-packages/agent/src/db-agent/tools/schemaDesignTool.ts

```diff
@@ -30,23 +106,80 @@ export const schemaDesignTool = tool(
       return `Input validation failed: ${errorDetails}. Please check your operations format and ensure all required fields are provided correctly.`
     }
 
-    const result = await repositories.schema.createVersion({
+    // Get current schema to validate DDL before creating version
+    const schemaResult = await repositories.schema.getSchema(designSessionId)
+    if (schemaResult.isErr()) {
+      // LangGraph tool nodes require throwing errors to trigger retry mechanism
+      // eslint-disable-next-line no-throw-error/no-throw-error
+      throw new Error(
+        'Could not retrieve current schema for DDL validation. Please check the schema ID and try again.',
+      )
+    }
+
+    // Apply operations to current schema to get the updated schema
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    const currentSchema = structuredClone(schemaResult.value.schema) as Schema
+    applySchemaOperations(currentSchema, parsed.output.operations)
+
+    // Validate DDL by generating and executing it
+    const { ddlStatements, results } = await validateAndExecuteDDL(
+      currentSchema,
+      designSessionId,
+    )
+
+    // Log the validation query and results
+    const queryResult = await repositories.schema.createValidationQuery({
+      designSessionId,
+      queryString: ddlStatements,
+    })
+
+    if (queryResult.success) {
+      await repositories.schema.createValidationResults({
+        validationQueryId: queryResult.queryId,
+        results,
+      })
+
+      // Log successful DDL execution to timeline
+      const successfulStatements = results.filter(
+        (result) => result.success,
+      ).length
+      const totalStatements = results.length
+      const summary = `DDL validation successful: ${successfulStatements}/${totalStatements} statements executed successfully`
+
+      // Create timeline item for DDL execution results
+      const result = await repositories.schema.createTimelineItem({
+        designSessionId,
+        content: summary,
+        type: 'query_result',
+        queryResultId: queryResult.queryId,
+      })
```

Comment:
Apparently, the timeline item is saved as a Human Message.I think this is a display issue and will fix it separately.

<img width="1388" height="1121" alt="Pasted_Image_2025_07_29__12_15" src="https://github.com/user-attachments/assets/c4b145d8-80da-4371-bbbc-aff1141cd11d" />